src/fonduer/parser/preprocessors/doc_preprocessor.py

```diff
@@ -26,29 +26,31 @@ def generate(self):
         """
         doc_count = 0
         for fp in self.all_files:
-            file_name = os.path.basename(fp)
-            if self._can_read(file_name):
-                for doc, text in self.parse_file(fp, file_name):
-                    yield doc, text
-                    doc_count += 1
-                    if doc_count >= self.max_docs:
-                        return
+            for doc in self.get_docs_for_path(fp):
+                yield doc
+                doc_count += 1
+                if doc_count >= self.max_docs:
+                    return
 
-    def parse_by_index(self, index):
+    def __getitem__(self, index):
         fp = self.all_files[index]
-        file_name = os.path.basename(fp)
-        if self._can_read(file_name):
-            for doc, text in self.parse_file(fp, file_name):
-                yield doc, text
+        return [doc for doc in self.get_docs_for_path(fp)]
```

Comment:
Is this correct? `__getitem__` returns all the documents, rather than one?

Reply:
Thanks! This is not used anymore and was wrong in its current form.